src/Store/ExtensionUtility.re

```diff
@@ -0,0 +1,347 @@
+/*
+ * StoreThread.re
+ *
+ * This is the 'state management' piece of Oni2.
+ *
+ * The state updates are run in a parallel thread to the rendering,
+ * so that we can eek out as much perf as we can in this architecture.
+ */
+
+open Revery;
+
+module Core = Oni_Core;
+module Option = Core.Utility.Option;
+
+module Extensions = Oni_Extensions;
+module Model = Oni_Model;
+
+open Oni_Extensions;
+
+module Log = (val Core.Log.withNamespace("Oni2.StoreThread"));
+module DispatchLog = (
+  val Core.Log.withNamespace("Oni2.StoreThread.dispatch")
+);
+
+let discoverExtensions = (setup: Core.Setup.t, cli: option(Core.Cli.t)) => {
+  open Core.Cli;
+  let extensions =
+    Core.Log.perf("Discover extensions", () => {
+      let extensions =
+        ExtensionScanner.scan(
+          // The extension host assumes bundled extensions start with 'vscode.'
+          ~prefix=Some("vscode"),
+          setup.bundledExtensionsPath,
+        );
+
+      let developmentExtensions =
+        switch (setup.developmentExtensionsPath) {
+        | Some(p) => ExtensionScanner.scan(p)
+        | None => []
+        };
+
+      let overriddenExtensionsDir =
+        cli |> Option.bind(cli => cli.overriddenExtensionsDir);
+
+      let userExtensions =
+        (
+          switch (overriddenExtensionsDir) {
+          | Some(p) => Some(p)
+          | None =>
+            switch (Core.Filesystem.getExtensionsFolder()) {
+            | Ok(p) => Some(p)
+            | Error(msg) =>
+              Log.errorf(m =>
+                m("Error discovering user extensions: %s", msg)
+              );
+              None;
+            }
+          }
+        )
+        |> Option.map(p => {
+             Log.infof(m => m("Searching for user extensions in: %s", p));
+             p;
+           })
```

Comment:
I often make a `tap` function that's useful for logging in the middle of expressions:

```reason
let tap = (f, x) => {
  f(x);
  x
};
```
Haven't gotten around to it here yet, but maybe you'd want to add it.

Reply:
Cool idea! Added it in our utilities